infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/InSnapshotSyncState.java

```diff
@@ -140,10 +143,15 @@ public LogReplicationState processEvent(LogReplicationEvent event) throws Illega
                 cancelSnapshotSync("replication terminated.");
                 return fsm.getStates().get(LogReplicationStateType.ERROR);
             default: {
-                log.warn("Unexpected log replication event {} when in snapshot sync state.", event.getType());
+                if (!fsm.isValidTransition(transitionEventId, event.getMetadata().getRequestId())) {
+                    log.warn("Ignoring log replication event {} for sync {} when in wait snapshot sync apply state for sync {}",
```

Comment:
Can an expected event reach in default case? Earlier, default case meant an unexpected event. If yes, could you please add a comment explaining this as well.